electricitymap/contrib/lib/models/events.py

```diff
@@ -220,17 +195,14 @@ def merge(cls, storage_mixes: List["StorageMix"]) -> "StorageMix":
         Merge a list of storage mixes into a single storage mix.
         The values are summed.
         """
-        storage_mix = cls()
+        merged_storage_mix = cls()
         for storage_mix_to_merge in storage_mixes:
-            for mode in STORAGE_MODES:
+            for mode in set(STORAGE_MODES).intersection(
+                storage_mix_to_merge.__fields_set__
+            ):
                 value = getattr(storage_mix_to_merge, mode)
-                if value is not None:
-                    current_value = getattr(storage_mix, mode)
-                    if current_value is None:
-                        storage_mix.set_value(mode, value)
-                    else:
-                        storage_mix.set_value(mode, current_value + value)
-        return storage_mix
+                merged_storage_mix.add_value(mode, value)
+        return merged_storage_mix
```

Comment:
Unless I am mistaken this is now the same (functionality wise) as the production merge method and both can be moved to the top level Mix class?

(or at least this can, since we are overriding the add_value() in the production mix to make sure it's not negative)

Reply:
I don't think it would because there's a slight difference in the production mixes where we copy the corrected modes set over to the new merged mix.
I would maybe keep as it is in this PR and give it another thought in another PR to keep this one small and focused on one change.